### ISSUE TITLE
BlenderBIM: Getting more usage out of Schedule Files (Sync'd Animations)

### ISSUE THREAD

**Issue by garylzimmer** [IfcOpenShell/IfcOpenShell#2050]
I would like to see eventually the ability to use the imported P6 schedules to create synchronized animated timelines that can be overlayed or appended to the side of a completed construction animation. 
[Something like this](https://youtu.be/4m27GUL2AXM)

Currently, it seems that the extent of the functionality is to produce an interactive Gantt chart html file.

**Comment by Moult** [comment]
So you can already generate an animation and it already shows a live date animated with it.

But no the animated timeline feature doesn't yet exist, is that what you're after?

**Comment by garylzimmer** [comment]
Yes. Animated timeline and maybe a text box field which can display any schedule tasks that are occurring at that point in time. Do you mind if I email you a recent animation video I did as an example?

**Comment by garylzimmer** [comment]
Is there anyway to use Python, Svorchek, or Geometry Nodes to access the loaded IFC work schedule?

I'm looking to programmatically create annotations from the work schedule. I have a proof of concept of this using Python to import CSV file with the layout I wanted but I was hoping I could merge that into the existing BlenderBIM features to use the formatting setup under the IFC Work Schedule panel.

**Comment by Moult** [comment]
```python
import blenderbim.tool
model = blenderbim.tool.Ifc.get()
print(model.by_type("IfcWorkSchedule"))
```

**Comment by BIMvoice** [comment]
@garylzimmer The core ask here (a synchronized animated 4D timeline with construction status colour coding, not just the interactive Gantt export) has been built since early-to-mid 2023: task bars (visual bar chart animated alongside the model), animation colour schemes per task type, an animation camera, and a live date overlay synced to the frame timeline. These are available today under the Work Schedule animation panel. Closing as implemented, feel free to reopen if there's a specific remaining gap (e.g. a text HUD listing active tasks by name, which isn't built yet).
